### ISSUE TITLE
Momentum problem (1-momentum is correct?) in BatchNorm2d

### ISSUE THREAD

**Issue by vikiQiu** [pytorch/pytorch#15298]
## 🐛 Bug

In BatchNorm2d' doc, the explanation of momentum is:
`\hat{x}_\text{new} = (1 - \text{momentum}) \times \hat{x} + \text{momemtum} \times x_t$`

where `x_t` is the newly computed batch statistics, `\hat{x}` is the estimated statistics in history.
While in my experiments, when momentum set to 0, the batchnorm operates just use the newly computed statistics.

So I guess maybe the following explanation is correct:

`\hat{x}_\text{new} = \text{momemtum\times \hat{x} + (1 - \text{momentum}) \times x_t$`

## To Reproduce

`b_x` is a torch.Tensor with shape (8, 3, 299, 299), which read from an image
`mol` is the model load from `Inception V3`
I have set the momentum as 0 in [Inception v3 model struct](https://github.com/pytorch/vision/blob/master/torchvision/models/inception.py):

```python
class BasicConv2d(nn.Module):

    def __init__(self, in_channels, out_channels, **kwargs):
        super(BasicConv2d, self).__init__()
        self.conv = nn.Conv2d(in_channels, out_channels, bias=False, **kwargs)
        self.bn = nn.BatchNorm2d(out_channels, eps=0.001, momentum=0)

    def forward(self, x):
        x = self.conv(x)
        x = self.bn(x)
        return F.relu(x, inplace=True)
```

Then I calculate the mean and var:
``` python
conv = mol.Conv2d_1a_3x3.conv(b_x)
dim = conv.shape
m = (conv.sum(0).sum(1).sum(1)/(dim[0]*dim[2]*dim[3])).view(1, -1, 1, 1)
var = (((conv-m)**2).sum(0).sum(1).sum(1)/(dim[0]*dim[2]*dim[3]-1)).view(1, -1, 1, 1) + 0.001
tmp = (conv - m) / torch.sqrt(var)
tmp = tmp * mol.Conv2d_1a_3x3.bn.weight.view(1, -1, 1, 1) + mol.Conv2d_1a_3x3.bn.bias.view(1, -1, 1, 1)
bn = mol.Conv2d_1a_3x3.bn(conv)
print((bn-tmp).mean(), (bn-tmp).std())
```

The result is that `bn=tmp` while `momentum = 0`

## Expected behavior

Expected `bn=tmp` while `momentum = 1`

## Environment

 - PyTorch Version 0.4.1:
 - Windows:
 - pip:
 - Python version: 3.6.5

**Comment by SsnL** [comment]
momentum has no effect on training mode output though.
